srcCpp/perftest_publisher.cxx

```diff
@@ -1702,7 +1734,8 @@ int perftest_cpp::Publisher()
     // indicating that it has discovered every Publisher
     fprintf(stderr,"Waiting for subscribers announcement ...\n");
     fflush(stderr);
-    while (_NumSubscribers > announcement_reader_listener->announced_subscribers) {
+    while (_NumSubscribers
+            > (int)announcement_reader_listener->subscriber_list.size()) {
```

Comment:
This is the first usecase: We are waiting until we receive the information of all the readers (we know we must receive _NumSubscribers number of them.